storage/cache/subtree_cache.go

```diff
@@ -424,7 +441,9 @@ func PopulateLogSubtreeNodes(treeHasher merkle.TreeHasher) storage.PopulateSubtr
 					// TODO(al): Don't panic Mr. Mainwaring.
 					panic(err)
 				}
-				if depth > 0 {
+				// Don't put leaves into the internal map and only update if we're rebuilding internal
```

Comment:
I'm not sure I follow the comment - isn't this block going to update the internal map?